File: c/isogram/isogram.c
#include "isogram.h"
#include <ctype.h>

bool is_isogram(const char phrase[])
{
    unsigned int alphabet = 0u;
    
    if (!phrase)
        return false;

    for (int i = 0; ; ++i)
    {
        char c = tolower(phrase[i]);
        if (c == '\0')
            break;
        if (c == ' ' || c == '-')
            continue;
        unsigned int flag = 1u << (c - 'a'); 
        if (alphabet & flag)
            return false;
        alphabet |= flag;
    }
    
    return true;
}

File: python/isogram/isogram.py
def is_isogram(string):
    """Returns if string is an isogram."""

    flag = 0b11111111111111111111111111
    state = 0
    for char in string:
        char = ord(char.lower()) - ord('a')
        if not 0 <= char < 26:
            continue
        position = 1 << char
        if state & position:
            return False
        state |= position
        
    return True